main/test/flix/Test.Exp.Let.Rec.flix

```diff
@@ -58,6 +58,45 @@ mod Test.Exp.LetRec {
         def loop(n) = if (n == 0) inc(0) else loop(dec(n));
         loop(10) == 1
 
+    @test
+    def testLetRecCapture05(): Bool =
+        let a = 1;
+        let b = 2;
+        def f(c) = if (c == 2) () -> c + b else if (c == 1) f(b) else f(a);
+        f(10)() == 4
+
+    @test
+    def testLetRecCapture06(): Bool =
+        let a = 1;
+        let b = 2;
+        def f(c) = if (c == 2) () -> c + b else if (c == 1) f(b) else f(a);
+        ((q) -> f(q))(10)() == 4
+
+    @test
+    def testLetRecCapture07(): Bool =
+        let a = 1;
+        let b = 2;
+        def f(c) = if (c == 2) () -> c + b else if (c == 1) f(b) else f(a);
+        (() -> f)()(10)() == 4
+
+    @test
+    def testLetRecCapture08(): Bool =
+        let a = 1;
+        let b = 2;
+        def f(c) = if (c == 2) () -> c + b else if (c == 1) f(b) else f(a);
+        (() -> f(b))()() == 4
+
+    @test
+    def testLetRecCapture09(): Bool =
+        let a = 1;
+        let b = 2;
+        let g = {
+            let d = 3;
+            def f(l) = if (l == 4) () -> d + l else f(a + b + 1);
+            () -> f
+        };
+        g()(5)() == 7
+
```

Comment:
What about a def capturing another def?

What about a def with a nested def?

Reply:
They are both tested above and below these tests, respectively :)